sympy/core/tests/test_function.py

```diff
@@ -722,15 +725,14 @@ def test_issue_7068():
 def test_issue_7231():
     from sympy.abc import a
     ans1 = f(x).series(x, a)
-    _xi_1 = ans1.atoms(Dummy).pop()
-    res = (f(a) + (-a + x)*Subs(Derivative(f(_xi_1), _xi_1), (_xi_1,), (a,)) +
-           (-a + x)**2*Subs(Derivative(f(_xi_1), _xi_1, _xi_1), (_xi_1,), (a,))/2 +
-           (-a + x)**3*Subs(Derivative(f(_xi_1), _xi_1, _xi_1, _xi_1),
-                            (_xi_1,), (a,))/6 +
-           (-a + x)**4*Subs(Derivative(f(_xi_1), _xi_1, _xi_1, _xi_1, _xi_1),
-                            (_xi_1,), (a,))/24 +
-           (-a + x)**5*Subs(Derivative(f(_xi_1), _xi_1, _xi_1, _xi_1, _xi_1, _xi_1),
-                            (_xi_1,), (a,))/120 + O((-a + x)**6, (x, a)))
+    res = (f(a) + (-a + x)*Subs(Derivative(f(y), y), y, a) +
```

Comment:
Something strange is going on here. After reading `res` from these lines in a debugger, I get
```
(Pdb) p res
f(a) + (-a + x)*Subs(Derivative(f(_xi_1), _xi_1), _xi_1, a) + (-a + x)**2*Subs(Derivative(f(y), (y, 2)), y, a)/2 + (-a + x)**3*Subs(Derivative(f(y), (y, 3)), y, a)/6 + (-a + x)**4*Subs(Derivative(f(y), (y, 4)), y, a)/24 + (-a + x)**5*Subs(Derivative(f(y), (y, 5)), y, a)/120 + O((-a + x)**6, (x, a))
```
Is there some explanation for the appearance of `_xi_1`?

Reply:
The dummy symbol is introduced by `Derivative._eval_derivative`. The result here is the same as the old test. Perhaps you are wondering why it is now written with `y` instead of `_xi_1`? The latter is a Dummy and fortunately, `Subs.__eq__` knows how to ignore those differences, so it doesn't matter what variable you use for the dummy variable of substitution. I used `y`.

Reply:
Why does `_xi_1` appear in the first order derivative only? Would it be possible to make that `y` as well? (I think it looks better.)

Reply:
I suspect this is related to issue #15138. What you see printed depends on what got stored in the cache. This raises an interesting situation, then: if it returns what is stored in the cache then do you ever have control over what is printed? A hack that comes to mind would be to have the printer clear the cache for that expression before printing. Can that limited amount of clearing be done?

Reply:
I made modifications in #15049 that allows one to change a symbol to another symbol that is not already in the expression so `Subs(x, x, 1).subs(x, y) -> Subs(y, y, 1)`.